js/main.js

```diff
@@ -357,56 +351,88 @@ function Search(a) {
         var a = b.input.val();
         b.searching && a == b.lastSearch ? b.close() : (b.state.addClass("searching"), b.search(a))
     });
+    
     this.dom = a;
+    
     this.close = function () {
         this.state.removeClass("searching");
         this.results.hide();
         this.searching = !1;
-        this.input.val("");//SAH -- let's erase string when we close
+        this.input.val("");
         nodeNormal()
     };
+    
     this.clean = function () {
         this.results.empty().hide();
         this.state.removeClass("searching");
         this.input.val("");
     };
+
     this.search = function (a) {
         var b = !1,
             c = [],
             b = this.exactMatch ? ("^" + a + "$").toLowerCase() : a.toLowerCase(),
             g = RegExp(b);
+
+        // Determine the type of the variable for which we're searching 
+        var exampleValue;
+
+        var firstElem;
+        for(elem in sigInst._core.graph.nodesIndex){
+            firstElem = elem;
+            break;
+        }
+        if(isNaN(firstElem))
+            firstElem = String(firstElem);
+        var exampleNode = sigInst._core.graph.nodesIndex[firstElem];
+
```

Comment:
Probably change it to:
var exampleNode;
for(elem in sigInst._core.graph.nodesIndex){
            firstElem = elem;
           if(isNaN(elem)) {
               elem = String(elem)
           }
          exampleNode = sigInst._core.graph.nodesIndex[elem];
          break;
 }